### PR TITLE
[Props] Add base properties test

### PR DESCRIPTION
**Pull request by zhoonit**
- [Test/Props] Add basic property tests 

```
This patch adds basic property test to demonstrate specification and
cover some test cases

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

- [Props] Add base properties 

```
Add base properties to be used for the property implementation

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: test/unittest/unittest_properties.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+  { /**< tag_cast does not have appropriate candidates */
+    using T = nntrainer::tag_cast<banana_prop_tag, int, std::string>::type;
+    ::testing::StaticAssertTypeEq<T, banana_prop_tag>();
+  }
+}
+
+TEST(BasicProperty, valid_p) {
+  { /** set -> get / to_string */
+    NumBanana b;
```
Comment (kparichay) [inline test/unittest/unittest_properties.cpp:65]: does this mean that every custom layer when they want new properties will have to make new class like numBanana or qualityOfBanana?

Comment (zhoonit) [reply test/unittest/unittest_properties.cpp:65]: If it is common, layer dose not need to make a new class else yes.

File: test/unittest/unittest_properties.cpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file unittest_properties.h
+ * @date 09 April 2021
+ * @brief This file contains test and sepecification of properties
```
Comment (lhs8928) [inline test/unittest/unittest_properties.cpp:7]: typo

**Review by lhs8928 — APPROVED** [review]
LGTM

File: test/unittest/unittest_properties.cpp (hunk 1 above)
Comment (kparichay) [reply test/unittest/unittest_properties.cpp:65]: So, all common properties must be opened to devel.

File: test/unittest/unittest_properties.cpp (hunk 2 above)
Comment (zhoonit) [reply test/unittest/unittest_properties.cpp]: Fixed

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**